src/structures/GuildPreviewEmoji.js

```diff
@@ -14,12 +14,12 @@ class GuildPreviewEmoji extends BaseGuildEmoji {
    */
 
   /**
-   * Set of roles this emoji is active for
-   * @type {Set<Snowflake>}
+   * The roles this emoji is active for
+   * @type {Snowflake[]}
    * @readonly
    */
   get roles() {
-    return new Set(this._roles);
+    return this._roles;
```

Comment:
I say we should return a slice... So people don't modify the value

Reply:
I think I actually prefer cloning the array too, but why should it still be kept as readonly then? Wouldn't that just make working with it unnecessarily harder?

Reply:
`Guild#features` is also not a getter, but a property. Since this is a getter, it's readonly (can only read, not write technically), but because we don't return the slice of the array it's "readonly but only to the class, not overall" which truly makes this confusing. I suggest we bring this in line with `Guild#features`

Reply:
> `Guild#features` is also not a getter, but a property. Since this is a getter, it's readonly (can only read, not write technically)

The property on the class is readonly, yes, but it's also marked as that with the current changes:
```ts
public readonly roles: Snowflake[];
          ^
```
Adding another `readonly` there would mean that it is frozen or cannot be modified because of other reasons, which isn't really true. If it returned a clone instead, there wouldn't be any problems with the user modifying it so I don't see the point of it there

<br>

> I suggest we bring this in line with `Guild#features`, if possible

Bring in line how exactly? Sorry if I'm not following